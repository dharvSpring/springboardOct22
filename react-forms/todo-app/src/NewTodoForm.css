.NewTodoForm {
    margin: 1rem;
}

.NewTodoForm > input {
    margin: 1rem;
}