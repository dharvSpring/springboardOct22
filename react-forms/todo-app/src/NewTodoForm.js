import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import './NewTodoForm.css';

function NewTodoForm({handleSubmit}) {
    const initialValues = {todoText: ''};
    let [formData, setFormData] = useState(initialValues);

    const handleChange = (evt) => {
        const {name, value} = evt.target;
        setFormData({
            ...formData,
            [name]: value,
        });
    }

    const submitTodo = (evt) => {
        evt.preventDefault();
        handleSubmit({
            ...formData,
            id: uuid(),
        });
        setFormData(initialValues);
    }
    return (
        <form className="NewTodoForm" onSubmit={submitTodo}>
            <label htmlFor="todoText">New Todo</label>
            <input type="text" id="todoText" name="todoText" value={formData.todoText} onChange={handleChange} />
            <input type="submit" value="Add Todo!" />
        </form>
    )
}

export default NewTodoForm;