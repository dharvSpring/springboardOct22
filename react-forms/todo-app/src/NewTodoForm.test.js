import React from "react";
import { render, fireEvent } from "@testing-library/react";

import NewTodoForm from "./NewTodoForm";

it("smoketest", function() {
    render(<NewTodoForm />);
});

it("snapshot", function() {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("calls create on submit", function () {
    const createFnMock = jest.fn();
    const { getByText } = render(<NewTodoForm handleSubmit={createFnMock} />);
    
    const addButton = getByText("Add Todo!");
    fireEvent.click(addButton);
    expect(createFnMock).toHaveBeenCalled();
});