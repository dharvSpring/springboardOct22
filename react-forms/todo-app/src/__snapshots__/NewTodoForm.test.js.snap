// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="todoText"
    >
      New Todo
    </label>
    <input
      id="todoText"
      name="todoText"
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Add Todo!"
    />
  </form>
</DocumentFragment>
`;
